local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -181,12 +181,22 @@ func (d *Discoverer) getVolumeType(fullPath string) (string, error) {
 	if isdir {
 		return common.VolumeTypeFile, nil
 	}
+	// check for Block before returning errdir
 	isblk, errblk := d.VolUtil.IsBlock(fullPath)
 	if isblk {
 		return common.VolumeTypeBlock, nil
 	}
 
-	return "", fmt.Errorf("Block device check for %q failed: DirErr - %v BlkErr - %v", fullPath, errdir, errblk)
+	// report the first error found
+	err := errdir
+	if err == nil {
+		err = errblk
+	}
+	if err != nil {
+		return "", fmt.Errorf("Error checking file %q: %s", fullPath, err)
+	}
+
+	return "", fmt.Errorf("Skipping file %q: not a directory nor block device", fullPath)
```

Comment:
nit: /nor/or/

Reply:
How about flipping the logic around...

if errdir ==nil && errblk == nil, then return "skipping"

and leaving the existing error message for both direrr and blkerr?